File: Server/src/main/content/global/skill/production/cooking/recipe/pie/impl/SummerPie.java
package content.global.skill.production.cooking.recipe.pie.impl;

import content.global.skill.production.cooking.recipe.pie.PieRecipe;
import core.game.node.item.Item;

/**
 * The Summer pie.
 */
public class SummerPie extends PieRecipe {

    private static final Item UNCOOKED_PIE = new Item(7216);

    private static final Item STRAWBERRY = new Item(5504);

    private static final Item WATERMELON = new Item(5982);

    private static final Item COOKING_APPLE = new Item(1955);

    private static final Item PART_ONE = new Item(7212);

    private static final Item PART_TWO = new Item(7214);

    @Override
    public Item getProduct() {
        return UNCOOKED_PIE;
    }

    @Override
    public Item[] getIngredients() {
        return new Item[]{STRAWBERRY, WATERMELON, COOKING_APPLE};
    }

    @Override
    public Item[] getParts() {
        return new Item[]{PIE_SHELL, PART_ONE, PART_TWO, UNCOOKED_PIE};
    }

    @Override
    public boolean isSingular() {
        return false;
    }

}

File: Emulator/src/main/content/global/skill/cooking/type/pie/impl/SummerPie.kt
package content.global.skill.cooking.type.pie.impl

import content.global.skill.cooking.type.pie.PieRecipe
import org.rs.consts.Items
import core.game.node.item.Item

/**
 * Summer pie.
 */
class SummerPie : PieRecipe() {

    override fun getProduct(): Item {
        return UNCOOKED_PIE
    }

    override fun getIngredients(): Array<Item> {
        return arrayOf(STRAWBERRY, WATERMELON, COOKING_APPLE)
    }

    override fun getParts(): Array<Item> {
        return arrayOf(PIE_SHELL, PART_ONE, PART_TWO, UNCOOKED_PIE)
    }

    override fun isSingular(): Boolean {
        return false
    }

    companion object {
        private val UNCOOKED_PIE = Item(Items.RAW_SUMMER_PIE_7216)
        private val STRAWBERRY = Item(Items.STRAWBERRY_5504)
        private val WATERMELON = Item(Items.WATERMELON_5982)
        private val COOKING_APPLE = Item(Items.COOKING_APPLE_1955)
        private val PART_ONE = Item(Items.PART_SUMMER_PIE_7212)
        private val PART_TWO = Item(Items.PART_SUMMER_PIE_7214)
    }
}
